pkg/geo/geomfn/shared_paths_test.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2020 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package geomfn
+
+import (
+	"testing"
+
+	"github.com/cockroachdb/cockroach/pkg/geo"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSharedPaths(t *testing.T) {
+	type args struct {
+		a *geo.Geometry
+		b *geo.Geometry
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    *geo.Geometry
+		wantErr bool
+	}{
+		{
+			name: "shared path between a multilinestring and linestring",
+			args: args{
+				a: geo.MustParseGeometry("MULTILINESTRING((26 125,26 200,126 200,126 125,26 125)," +
+					"(51 150,101 150,76 175,51 150))"),
+				b: geo.MustParseGeometry("LINESTRING(151 100,126 156.25,126 125,90 161, 76 175)"),
+			},
+			want: geo.MustParseGeometry("GEOMETRYCOLLECTION(MULTILINESTRING((126 156.25,126 125)," +
+				"(101 150,90 161),(90 161,76 175)),MULTILINESTRING EMPTY)"),
+		},
+		{
+			name: "shared path between a linestring and multilinestring",
+			args: args{
+				a: geo.MustParseGeometry("LINESTRING(76 175,90 161,126 125,126 156.25,151 100)"),
+				b: geo.MustParseGeometry("MULTILINESTRING((26 125,26 200,126 200,126 125,26 125), " +
+					"(51 150,101 150,76 175,51 150))"),
+			},
+			want: geo.MustParseGeometry("GEOMETRYCOLLECTION(MULTILINESTRING EMPTY," +
+				"MULTILINESTRING((76 175,90 161),(90 161,101 150),(126 125,126 156.25)))"),
+		},
```

Comment:
can you add a case for no shared paths here?

Reply:
working on this. facing an issue with empty geomtrycollection. the actual and the expected length of the EWKB differs. checking this.. 

```
        shared_paths_test.go:77:
            	Error Trace:	shared_paths_test.go:77
            	Error:      	Not equal:
            	            	expected: &geo.Geometry{spatialObject:geopb.SpatialObject{Type:2, EWKB:geopb.EWKB{0x1, 0x7, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0}, SRID:0, ShapeType:8, BoundingBox:(*geopb.BoundingBox)(nil)}}
            	            	actual  : &geo.Geometry{spatialObject:geopb.SpatialObject{Type:2, EWKB:geopb.EWKB{0x1, 0x7, 0x0, 0x0, 0x0, 0x2, 0x0, 0x0, 0x0, 0x1, 0x5, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1, 0x5, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0}, SRID:0, ShapeType:8, BoundingBox:(*geopb.BoundingBox)(nil)}}

            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -3,4 +3,5 @@
            	            	   Type: (geopb.SpatialObjectType) 2,
            	            	-  EWKB: (geopb.EWKB) (len=9) {
            	            	-   00000000  01 07 00 00 00 00 00 00  00                       |.........|
            	            	+  EWKB: (geopb.EWKB) (len=27) {
            	            	+   00000000  01 07 00 00 00 02 00 00  00 01 05 00 00 00 00 00  |................|
            	            	+   00000010  00 00 01 05 00 00 00 00  00 00 00                 |...........|
            	            	   },
```

Reply:
looks like
```

otan=# select st_astext('010700000002000000010500000000000000010500000000000000'::geometry);
                            st_astext
-----------------------------------------------------------------
 GEOMETRYCOLLECTION(MULTILINESTRING EMPTY,MULTILINESTRING EMPTY)
(1 row)

```